app/models/district.rb

```diff
@@ -10,26 +10,51 @@
 #
 
 class District < ActiveRecord::Base
-  belongs_to :state
+  belongs_to :state, required: true
+  has_many :senators, through: :state
   has_and_belongs_to_many :zip_codes
   has_one :representative, class_name: "Legislator"
+  has_one :target_rep, -> { targeted }, class_name: "Legislator"
   has_many :campaigns, through: :representative
 
-  validates :state, presence: true
   validates :district, uniqueness: { scope: :state }
 
   def self.find_by_state_and_district(state:, district:)
     joins(:state).where('states.abbrev': state).find_by(district: district)
   end
 
+  def self.find_by_address(address:, zip:, city: nil, state: nil, includes: nil)
+    results = Integration::Here.geocode_address( address: address,
+                                                 city:    city,
+                                                 state:   state,
+                                                 zip:     zip )
+    if coords = results[:coordinates]
+      if district_hash = Integration::MobileCommons.district_from_coords(coords)
+        District.includes(includes).find_by_state_and_district(district_hash)
+      end
+    end
+  end
+
   def fetch_rep
     self.representative = Legislator.fetch_one(district: self)
   end
 
+  def unconvinced_rep
+    representative unless representative.with_us?
+  end
+
+  def unconvinced_legislators
+    senators.eligible.unconvinced + [unconvinced_rep].compact
```

Comment:
compact just removes the nil if there's no unconvinced_rep

without compact it's either `senators + [rep]` or `senators + [nil]`

with compact it's `senators + []` which leaves senators unchanged